internal/provisioner/cluster_installation_provisioner.go

```diff
@@ -1048,53 +1048,41 @@ func getMattermostEnvWithOverrides(installation *model.Installation) model.EnvVa
 }
 
 // getIngressAnnotations returns ingress annotations used by Mattermost installations.
-func getIngressAnnotations() map[string]string {
-	return map[string]string{
-		"kubernetes.io/tls-acme":                               "true",
-		"nginx.ingress.kubernetes.io/proxy-buffering":          "on",
-		"nginx.ingress.kubernetes.io/proxy-body-size":          "100m",
-		"nginx.ingress.kubernetes.io/proxy-send-timeout":       "600",
-		"nginx.ingress.kubernetes.io/proxy-read-timeout":       "600",
-		"nginx.ingress.kubernetes.io/proxy-max-temp-file-size": "0",
-		"nginx.ingress.kubernetes.io/ssl-redirect":             "true",
-		"nginx.ingress.kubernetes.io/configuration-snippet": `
-				  proxy_force_ranges on;
-				  add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;`,
-		"nginx.org/server-snippets": "gzip on;",
-	}
+func getIngressAnnotations() *model.IngressAnnotations {
+	annotations := &model.IngressAnnotations{}
+	annotations.SetDefaults()
+	return annotations
 }
 
 // getHibernatingIngressAnnotations returns ingress annotations used by
 // hibernating Mattermost installations.
-func getHibernatingIngressAnnotations() map[string]string {
-	return map[string]string{
-		"nginx.ingress.kubernetes.io/configuration-snippet": "return 410;",
-	}
+func getHibernatingIngressAnnotations() *model.IngressAnnotations {
+	annotations := &model.IngressAnnotations{}
+	annotations.SetHibernatingDefaults()
+	return annotations
 }
 
-func addInternalSourceRangesToAnnotations(annotations map[string]string, allowedIPRanges *model.AllowedIPRanges, internalIPRanges string) {
-	if allowedIPRanges == nil || len(*allowedIPRanges) == 0 {
+func addSourceRangeWhitelistToAnnotations(annotations *model.IngressAnnotations, allowedIPRanges *model.AllowedIPRanges, internalIPRanges []string) {
+	// If all allowedIPRanges are disabled we don't continue so we don't unintentionally lock the workspace to internal traffic only
+	if allowedIPRanges == nil || len(*allowedIPRanges) == 0 || allowedIPRanges.AllRulesAreDisabled() {
```

Comment:
I've removed the redundant nil check and added the len check to the `AllRulesAreDisabled()` function.

We have to have an early exit here rather than going through the no-op for loop because while yes the user-set allowedIPRanges might be empty, the internal ones won't be, so we'd be effectively locking the installation to our own internal IP's only